bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/internal/handler/LGThinQAbstractDeviceHandler.java

```diff
@@ -0,0 +1,820 @@
+        isThingReconfigured = true;
+        if (getThing().getConfiguration().containsKey("pollingPeriodPowerOnSeconds")) {
+            pollingPeriodOnSeconds = ((BigDecimal) getThing().getConfiguration().get("pollingPeriodPowerOnSeconds"))
+                    .intValue();
+        }
+        if (getThing().getConfiguration().containsKey("pollingPeriodPowerOffSeconds")) {
+            pollingPeriodOffSeconds = ((BigDecimal) getThing().getConfiguration().get("pollingPeriodPowerOffSeconds"))
+                    .intValue();
+        }
+        if (getThing().getConfiguration().containsKey("pollingExtraInfoPeriodSeconds")) {
+            pollingExtraInfoPeriodSeconds = ((BigDecimal) getThing().getConfiguration()
+                    .get("pollingExtraInfoPeriodSeconds")).intValue();
+        }
+        if (getThing().getConfiguration().containsKey("pollExtraInfoOnPowerOff")) {
+            pollExtraInfoOnPowerOff = (Boolean) getThing().getConfiguration().get("pollExtraInfoOnPowerOff");
+        }
+        // if the periods are the same, I can define currentPeriod for polling right now. If not, I postpone to the nest
+        // snapshot update
+        if (pollingPeriodOffSeconds == pollingPeriodOnSeconds) {
+            currentPeriodSeconds = pollingPeriodOffSeconds;
+        }
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        getLogger().debug("bridgeStatusChanged {}", bridgeStatusInfo);
+        // restart scheduler
+        initializeThing(bridgeStatusInfo.getStatus());
+    }
+
+    /**
+     * Determine if the Handle for this device supports Energy State Collector
+     * 
+     * @return always false and must be overridden if the implemented handler supports energy collector
+     */
+    protected boolean isExtraInfoCollectorSupported() {
+        return false;
+    }
+
+    /**
+     * Returns if the energy collector is enabled. The handle that supports energy collection must
+     * provide a logic that defines if the collector is currently enabled. Normally, it uses a Switch Channel
+     * to provide a way to the user turn on/off the collector.
+     * 
+     * @return true if the energyCollector must be enabled.
+     */
+    protected boolean isExtraInfoCollectorEnabled() {
+        return false;
+    }
+
+    private class UpdateExtraInfoCollector implements Runnable {
+        @Override
+        public void run() {
+            updateExtraInfoState();
+        }
+    }
+
+    private void updateExtraInfoState() {
+        if (!isExtraInfoCollectorSupported()) {
+            logger.error(
+                    "The Energy Collector was started for a Handler that doesn't support it. It most likely a bug.");
+            return;
+        }
+        try {
+            Map<String, Object> extraInfoCollected = collectExtraInfoState();
+            updateExtraInfoStateChannels(extraInfoCollected);
+        } catch (LGThinqException ex) {
+            getLogger().error(
```

Comment:
Both `getLogger()` and `logger` are both used together. Better to be consistent and choose one to use everywhere
```suggestion
            logger.error(
```